security/medusa/l2/kobject_socket.c

```diff
@@ -15,14 +15,13 @@ MED_ATTRS(socket_kobject) {
 	MED_ATTR_END
 };
 
-int socket_kobj2kern(struct socket_kobject * sock_kobj, struct socket * sock)
+void socket_kobj2kern(struct socket_kobject *sock_kobj, struct socket *sock)
 {
 	struct medusa_l1_socket_s *sk_sec = sock_security(sock->sk);
 	sock_kobj->med_object = sk_sec->med_object;
-	return 0;
 }
```

Comment:
I mean implementation of this function is not valid.
However, each *kobj2kern and *kern2kobj function should return an success/failure code which must be always tested; see IPC submodule  (kobject_ipc.c) as an example.
I suggest leave (for now) signature unchanged, and ASAP I try rewrite using of all *kobj2kern and *kern2kobj functions as in IPC submodule.

Reply:
Can this return anything else except 0? I did not see much of a sense in returning anything in this example, since copying is a straightforward operation.

Reply:
implementation of this function seems not finished yet. so for now, there is no reason for returning some value, but there will be, one day ;)